### ISSUE TITLE
[mate] too many images?

### ISSUE THREAD

**Issue by udeved** [manjaro/manjaro-tools-iso-profiles#94]
I don't know mate, but this profile has many copies of the same background in different locations.
Is this necessary in mate? Can't be all this stuff set in dconf to use /usr/share/backgrounds?

The annoying part is, that dconf seems to be a binary blob, only accessible with dconf? How can I quickly see whats stored in the dconf in the profile?

**Comment by oberon2007** [comment]
can only be read/edited with dconf-editor by the user I'm afraid ... ;P

**Comment by udeved** [comment]
This binary dconf xml(?) blob is like MS groundhog day. I big step backward in open source, moving to closed source somehow.

**Comment by udeved** [comment]
@oberon2007 
do you have dconf installed, or mate?
I am totally confused how this can be edited from non mate? Impossible?

**Comment by oberon2007** [comment]
But there is a cleaner way: You can deliver your settings by ways of overrides in **/usr/share/glib-2.0/schemas/** and dconf will be created with ``sudo glib-compile-schemas /usr/share/glib-2.0/schemas`` which will happen during install anyway ...

**Comment by udeved** [comment]
Would you mind taking a look at the mate and sonar mate profiles and try to set the various background all to /usr/share/something and remove redundant images? These profiles are still too big in my view.

**Comment by oberon2007** [comment]
I can't do it right now. Maybe @coffeeking could do that for the sonar profile, or simply replace some of the extra images by symlinks? What about Ringo? Is he not on github?

**Comment by udeved** [comment]
I already asked ringo, but biggest struggle was to explain to him what I wanted to know. 
It wasn't really productive.

**Comment by udeved** [comment]
The sonar people all don't have sight, I thought it would be cool if we could migrate and optimize their profiles in our repo for them. They may find it big help, and then sonar mate is actually a customized copy of mate, hence both have this image size problem.

**Comment by Ste74** [comment]
Only a personal thing .. is possible download image and reduce the size (mb->kb)and reupload them or your intent is unified image ?

**Comment by udeved** [comment]
@Ste74 

I am asking, if all these background images that are used in mate profiles could be made to use /usr/share/backgrounds, and we just have one or two images instead of 4 similar?

**Comment by Ste74** [comment]
:) see now .. also same image , size different and location different ;)

**Comment by oberon2007** [comment]
main problem is that they are different format .png and .jpg Otherwise we could simply replace one by a symlink ... :P

**Comment by udeved** [comment]
yes, ugly, but simply opening with imagviewer and saving in jpg could do, if we were able to set the dconf without mate. Great, whoever had the idea to make dconf a binary blob was a inspired by win7.

**Comment by coffeeking** [comment]
hi
Re sonar, this is true. Both myself and kyle brouhard, the other main 
developer, both have some sight, but not much. We've been struggling for 
months now to find out why our mate images won't boot to the desktop, 
while the manjaro mate images do work. All sonar is at it's core is a 
version of manjaro wich talks. Same installer, or this will be the case 
once calamares becomes accessible, and all the packages I could think of 
that a blind person might need, audio converters, video converters, etc. 
The images are quite large, around 1.7 gb. I'd like to reduce this, but 
I seem to have a problem taking out software, grins. I would also like 
to turn on isolinux's talking functions, which are supposed to work by 
pressing f9 key. This would enable the live images to talk. But I can no 
longer remember how this is done. I'd gladly accept the help.
Thanks
Kendell clark

**Comment by coffeeking** [comment]
hi
I agree, dconf is ugly. But it's the only way I know of to set settings 
up in a live image. I am forced to do this to set our preset keyboard 
shortcuts, desktop  icons, and in the future, orca settings. There are 
gui viewers for this stuff but the ironic thing is that they are not 
very accessible.
Thanks
Kendell clark

**Comment by udeved** [comment]
Hi Kendell,

nice to hear from you.
I try to keep it short.

My idea is that you sonar people commit to the manjaro-iso-profiles repo.
I basically integrated your sonar profiles, they may need tweaking, but they build.
This may have good help effects for you, in case we update something, I can also make your profiles work then, just like I do with the rest.

As for isolinux, this is good you mention your needs, we thought about getting you your own isolinux package, we'd just have to customize the source gfxboot for sonar. manjaro-tools-iso will handle different isolinux providers with pacman.

Regards
Marcus(artoo)

**Comment by udeved** [comment]
Kendell

have you considered migrating to plasma? Any experience with accessibility and kde?

**Comment by coffeeking** [comment]
hi
This is wonderful! Yeah this was the eventual idea. Sonar is a separate 
project because it's not manjaro, but it's *only* a version of manjaro 
that talks. So our eventual goal was to be featured in the manjaro 
download page, with a little message heading, something like "if you are 
visually impaired, try ..." message. Neither kyle or I are experts in 
linux. I learn as I go, but to be completely blunt, dconf is an ugly 
mess, and I don't like it much. But it's what I have to work with. The 
best way I've found to make our live images come up talking with 
specific desktop icons and keyboard shortcuts is to copy everything into 
the mate overlay, in /etc/skel. It's ugly, but it does work. The main 
problem with this is I have literally no idea how to keep the images 
clean. I've already seen some commits by phil saying I've made our 
images look ugly with no idea how I did it. Our images won't even boot 
all the way and I've been tearing my hair out trying to figure out why. 
No missing files, login manager is properly set up, I'm stumped. At this 
point I'd like to replace lightdm with gdm, since it's more accessible. 
The main problem with that is that gdm pulls in too much of gnome shell. 
If we had a gdm minimal package, but I'm not even sure if this is 
possible. Too much of gnome depends on other bits of gnome.
Thanks
Kendell clark

**Comment by coffeeking** [comment]
hi
Plasma? I'd love to have a plasma edition. The main problem with this is 
that plasma is completely inaccessible. It should be accessible, as it's 
using qt5, but my initial testing with it, mind you this was several 
months ago when it was new was that orca wouldn't read any of it. I'll 
try this on my sonar box and see if this works better. What I really 
need is more help. I need sighted users who have the time using orca, 
which is the screen reader, on desktops they use and filing the bugs. 
I'm literally either the only or one of only a few blind people who 
bother to file bugs against these desktops and applications which aren't 
accessible. The rest just whine and compare linux to windows and opine. 
You know the type. I'd also like to make a cinnamon edition, which 
manjaro does have in it's repositories, but cinnamon is just not quite 
there yet. It almost works, but it's panel and menu don't work right, 
and if I gave a linux newbie an inaccessible desktop, they'd run back to 
windows about as fast as they possibly could. If you or anyone else, 
maybe @burnhard? could do this it would be a huge help. I'm literally 
overworked, at my own doing. I file bugs against orca 
<http://github.com/gnome/orca>, the mate desktop 
<http://github.com/mate-desktop> and gnome itself. The reception is ... 
good, but the people there are overworked as well.
Thanks
Kendell clark

**Comment by udeved** [comment]
Kendell

if we assume you start to commit and push to the manjaro iso profiles, I believe you will greatly profit from it.
I have made some heavy profile changes recently,and from development point of view, it helps to have the profiles all in one place. Some other changes made buildiso run anywhere in the filesystem, but the downside is, the repo name has to be change if its no default manjaro repo. This may be a big stumbling block for people without sight. 
We could also easily package your profiles too for the repos, and joining in one repo may attract also more help from community.

My idea would be, we slowly build up a sonar cli version first, which should serve as solid base for other desktop experiments. I may be able to find a special sonar solution, to keep maintainig package lists at minimum for you.
Once we have a solid cli version, we could make some testing experiments with different desktop environments and pick the best solution. I saw you were frustated and deleted your cli profile.

**Comment by coffeeking** [comment]
hi
This is a great idea. I'd like to return the cli profile. The idea was 
to make a basic version that came up talking. This should be easily 
done, but I couldn't actually make it talk. One of the biggest problems 
that needs to get fixed that I've been unable to manage is the problem 
of speakup and pulse audio. Speakup is a builtin kernel module, so must 
run as root. Pulse audio is a user process, unless run in system wide 
mode which can be done, but I haven't managed that either. So when 
speakup uses software text to speech programs like espeak, through the 
espeakup package, the sound runs as root, and pulse doesn't let it 
through. I've been trying to figure out whether this is a pulseaudio 
problem or a speakup one. Both communities are unwilling to fix it, 
passing the bug. Pulse says it's not their problem, speakup says they 
hate pulse audio, don't use it, etc. This doesn't impact a cli version, 
but once a user installs a desktop, suddenly speech will stop in the cli 
once that desktop is started. This should be fixed in whichever program 
is causing the issue, but as a temporary solution, I could try to make 
espeakup start *after* a user logs in, which will cause pulse audio to 
allow the sound through. I think this might be done by modifying the 
systemd service file to only start after multi-user.target. Failing 
that, maybe after a user logs in, which might be done by getty@*? I'm 
not at all sure, I'm still new to systemd. I'll gladly accept sonar 
packages. We already have a special brltty one, although I think that 
one might need to be eventually incorporated upstream in the arch 
package. Espeak, the main software tts we use has been forked into 
espeak-ng, since we're not sure if the main dev is dead or simply non 
responsive. I can upload a pkgbuild replacing the espeak one in 
manjaro's repos but this too should eventually make it upstream to other 
distros, arch, ubuntu, etc. I wouldn't have so much to do if I hadn't 
let myself get nearly a month behind when my wife went to her parents 
for the holidays, but once she left I sort of lost all motivation to do 
anything but wait by the phone and computer, lol
Thanks
Kendell clark

**Comment by udeved** [comment]
Hi Kendell,

my proposal for a kde based sonar was the thought, that instead of jumping through hoops with dconf, we try to build a kde with accessibility, because kde has plain text based configuration. Should be much easier for you to edit and customize.
I had a look yesterday at the porting status of the accessibility modules of kde, they are still in transition and not fully kf5 ported. But time plays for you here. 

Regards 
Marcus

**Comment by philmmanjaro** [comment]
I know that most people like to have KDE5 also accessible. For now it isn't yet. Gnome and Mate are the only two fully supported user interfaces. However we can always ask the maintainer of Budgie to support it if it isn't already. Dconf is the bad habit and Mate adopted it now. It is binary but I assume not so much must be changed, well when a new GTK version is out, you have to upload a new binary config file, but it was there from almost the beginning as getting from its [history](https://github.com/manjaro/manjaroiso/commits/master/configs/gnome/gnome-overlay/etc/skel/.config/dconf/user).

**Comment by philmmanjaro** [comment]
In regard of the background images here. Are they used anyway? Any background should be taken from a package and not added to the overlay. So the tasks to be done here:

* remove the pictures from the overlay
* add needed packages to the list
* adjust the dconf binary

**Comment by udeved** [comment]
Well, I also have one background image in overlay, its the green maia one renamed. 
I found the package that contains the background too big. 
Anyway, the size is small.

Perhaps we should say, if any image, please not eg 6MB png 3000x2000

What would be a sane max dimension? 1920x1080?

**Comment by philmmanjaro** [comment]
We have now followed situation. Community editions are the bigger profiles, not all of them off course:

```
816K	./community/budgie
824K	./community/i3
2,4M	./community/enlightenment
692K	./community/netbook
620K	./community/fluxbox
084K	./community/gnome
3,5M	./community/pantheon
288K	./community/deepin
188K	./community/kde-next
1,6M	./community/bspwm
212K	./community/cinnamon
512K	./community/lxqt
012K	./community/openbox
11,0M	./community/jwm
584K	./community/lxde
8,9M	./community/mate
020K	./community/sysmenu
32,0M	./community
```

Official profiles are the smallest, as they only hold minimum config files:

```
540K	./official/kde
032K	./official/net
724K	./official/xfce
1,3M	./official
```

Since **isolinux** is now separate, no issues with **shared**:

```
128K	./shared/overlay
3,0M	./shared/isolinux
116K	./shared/overlay-livecd
3,3M	./shared
```

Same goes for the minimal profiles:

```
880K	./minimal/budgie-minimal
512K	./minimal/kde-minimal
580K	./minimal/xfce-minimal
372K	./minimal/lxqt-minimal
040K	./minimal/net-minimal
2,4M	./minimal
```

The list is from the current **master** branch of 0.9.15.

Adding binaries in any way creates a massive git overlay:

```
084K	./.git/logs
08,0K	./.git/info
04,0K	./.git/branches
044K	./.git/hooks
117M	./.git/objects
072K	./.git/refs
117M	./.git
```

So I'd say that some is off with a profile, if it is larger than 1 mega byte. 11 MB for JWM is something we should also check ...

**Comment by udeved** [comment]
I reduced them in devel branch already.

The problematic profiles:

mate/sonar-mate
jwm, Holmes needs to tell where he downloaded some stuff so it can be packaged.

Current devel branch:

community 33.4 MB
minimal 1.7 MB
netrunner 650 kB
official 823 kB
sonar 7.5 MB
shared 511 kB

**Comment by philmmanjaro** [comment]
jwm is more or less themes in the overlay, which is a no-go. Sonar I've to check. Maybe something added which is not needed. I'd say we should make the switch for all given profiles and mark those we have to check in a todo list. Also I'd like to have a list of maintainers for each profile we can contact. And a list of no-gos or preferred way we want to have our profiles structured. Like a manual or readme. We simply have to document it better and review new profiles before adopting them on the given standards. Don't get me wrong, but sometimes we are to "soft" ;)

**Comment by udeved** [comment]
Additionally, bspwm and openbox in devel are too big too.

~~~
$ du -c -d 2
780     ./netrunner/kde-netrunner
372     ./netrunner/lxqt-netrunner
1156    ./netrunner
32      ./official/net
544     ./official/kde
740     ./official/xfce
1320    ./official
148     ./shared/netrunner
356     ./shared/sonar
264     ./shared/manjaro
800     ./shared
40      ./minimal/net-minimal
880     ./minimal/budgie-minimal
512     ./minimal/kde-minimal
580     ./minimal/xfce-minimal
356     ./minimal/lxqt-minimal
2372    ./minimal
36      ./.git/refs
4       ./.git/branches
44      ./.git/hooks
111108  ./.git/objects
8       ./.git/info
40      ./.git/logs
111612  ./.git
4180    ./sonar/gnome-sonar
4060    ./sonar/mate-sonar
8244    ./sonar
84      ./community/gnome
832     ./community/fluxbox
692     ./community/netbook
3524    ./community/pantheon
20      ./community/sysmenu
212     ./community/cinnamon
512     ./community/lxqt
5896    ./community/openbox
7804    ./community/mate
584     ./community/lxde
2376    ./community/enlightenment
4612    ./community/jwm
292     ./community/deepin
1200    ./community/i3
6940    ./community/bspwm
1372    ./community/budgie
36956   ./community
162476  .
162476  insgesamt
~~~

I agree, we need some standards to meet before profiles get adopted.

**Comment by oberon2007** [comment]
I have recently taken care of **openbox** and packed the icon-theme and font before I pushed it to devel.
The remaining problem is the gtk-theme: **evopop** has become a theme without openbox-support meanwhile, so it's not possible to just use the version we have in the repo. If we pack a separate evopop-openbox version well need to keep the sources somewhere else and change the config - which is something @holmeslinux  or whoever will maintain Openbox in the future would have to do. I have contacted him already but he seems to be unavailable right now ...

**Comment by philmmanjaro** [comment]
When we have the list of all current maintainers we will talk to them and point them to the stuff they should optimize. We should guide them rather then shout at them ;) It is not their mistake, we simply don't gave them the proper guidelines yet.

In general the development of **manjaro-tools** is great. It is a powerful set of nice tools which makes the development of Manjaro great and simple. Now we have to adopt this also to the profiles and direct our contributors.

**Comment by oberon2007** [comment]
Ah well, it's not true what I just said about openbox-config. We can make an evopop-openbox theme and install to the same folder as it is now kept in, then the config will not be a problem.

**Comment by philmmanjaro** [comment]
Creating a new repo for the theme is no problem. I simply want to have it out of this one.

**Comment by udeved** [comment]
Once we have the profiles sorted, we can start to move towards packages for the DE skeletons.
https://github.com/manjaro/desktop-settings

**Comment by mhussaincov** [comment]
hi there,
if kendle has not seen this message, will he be informed on the changes?
there is a gnome profile also in sonar I believe.

Majid

On 07/01/2016, udeved <notifications@github.com> wrote:
> Additionally, bspwm and openbox in devel are too big too.
>
> ~~~
> $ du -c -d 2
> 780     ./netrunner/kde-netrunner
> 372     ./netrunner/lxqt-netrunner
> 1156    ./netrunner
> 32      ./official/net
> 544     ./official/kde
> 740     ./official/xfce
> 1320    ./official
> 148     ./shared/netrunner
> 356     ./shared/sonar
> 264     ./shared/manjaro
> 800     ./shared
> 40      ./minimal/net-minimal
> 880     ./minimal/budgie-minimal
> 512     ./minimal/kde-minimal
> 580     ./minimal/xfce-minimal
> 356     ./minimal/lxqt-minimal
> 2372    ./minimal
> 36      ./.git/refs
> 4       ./.git/branches
> 44      ./.git/hooks
> 111108  ./.git/objects
> 8       ./.git/info
> 40      ./.git/logs
> 111612  ./.git
> 4180    ./sonar/gnome-sonar
> 4060    ./sonar/mate-sonar
> 8244    ./sonar
> 84      ./community/gnome
> 832     ./community/fluxbox
> 692     ./community/netbook
> 3524    ./community/pantheon
> 20      ./community/sysmenu
> 212     ./community/cinnamon
> 512     ./community/lxqt
> 5896    ./community/openbox
> 7804    ./community/mate
> 584     ./community/lxde
> 2376    ./community/enlightenment
> 4612    ./community/jwm
> 292     ./community/deepin
> 1200    ./community/i3
> 6940    ./community/bspwm
> 1372    ./community/budgie
> 36956   ./community
> 162476  .
> 162476  insgesamt
> ~~~
>
> I agree, we need some standards to meet before profiles get adopted.
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/manjaro/manjaro-tools-iso-profiles/issues/94#issuecomment-169734690


-- 
kind regards,
Majid Hussain

**Comment by udeved** [comment]
I also think, it would be much better if ringo was working in the manjaro repo, since he maintains mate and cinnamon I believe.

**Comment by philmmanjaro** [comment]
Hi Majid,

we have Kendell already informed about it. He is up to speed on this topic. We will create DE skeletons which can be used as packages. The profiles will be soon or later just simple constructs to select the right parts to build your install media. The goal is to reuse given parts similar like Lego bricks in the best way you see it fit.

This way we will have less overlay and multible files.

**Comment by philmmanjaro** [comment]
@udeved: is something still holding us back on doing the switch this weekend? As I can see it, it is already pretty settled by now ...

**Comment by udeved** [comment]
Nothing, we can do it I think.
However, it would be very cool, if we could manage to achieve smaller sizes until Saturday.
Ideally, we'd have smallest possible profiles, with all other packaged, and next phase in master would be to remove skeletons.

**Comment by philmmanjaro** [comment]
Since I might not be available by tomorrow we have to think about it. I suggest to do a list of these profiles built and tested. The bigger profiles we can check what is missing and hold them of from merging, for now. Simply create a branch in which we can work on them. When satisfied we can merge them into master. What do you think?

**Comment by udeved** [comment]
List sounds good, a task/assignment/todo list yes.

I don't mind merging as is in master either, I was just a bit daydreaming of the ideal. :laughing: 
Its in my view nothing of a blocker, we can also sort it out in master in combination with desktop-setting sub project.

**Comment by udeved** [comment]
For the sake of readability
~~~
$ du  -hcd 2  
780K    ./netrunner/kde-netrunner
372K    ./netrunner/lxqt-netrunner
1,2M    ./netrunner
32K     ./official/net
544K    ./official/kde
740K    ./official/xfce
1,3M    ./official
148K    ./shared/netrunner
356K    ./shared/sonar
264K    ./shared/manjaro
800K    ./shared
40K     ./minimal/net-minimal
880K    ./minimal/budgie-minimal
512K    ./minimal/kde-minimal
580K    ./minimal/xfce-minimal
356K    ./minimal/lxqt-minimal
2,4M    ./minimal
36K     ./.git/refs
4,0K    ./.git/branches
44K     ./.git/hooks
109M    ./.git/objects
8,0K    ./.git/info
40K     ./.git/logs
109M    ./.git
4,1M    ./sonar/gnome-sonar
4,0M    ./sonar/mate-sonar
8,1M    ./sonar
84K     ./community/gnome
832K    ./community/fluxbox
692K    ./community/netbook
3,5M    ./community/pantheon
20K     ./community/sysmenu
212K    ./community/cinnamon
512K    ./community/lxqt
5,8M    ./community/openbox
7,7M    ./community/mate
584K    ./community/lxde
2,4M    ./community/enlightenment
4,6M    ./community/jwm
292K    ./community/deepin
1,2M    ./community/i3
6,8M    ./community/bspwm
1,4M    ./community/budgie
37M     ./community
159M    .
159M    insgesamt

~~~

**Comment by philmmanjaro** [comment]
Well, I don't like to have a complete snapshot of a forked theme in our profiles. Openbox simply added [evopop](http://entornosgnulinux.com/2015/07/04/openbox-theme-evopop/) for gtk 3.16 to it. This is unmaintained and most likely broken with the current gtk 3.18. However bunsen has a [new theme](https://github.com/BunsenLabs/bunsen-themes).

**Comment by udeved** [comment]
#95 

I see, I just saw activity on that.
We can also say wait another week and sort it out in devel, I basically don't mind a "problem" branch either.

**Comment by oberon2007** [comment]
Have a look at this @philmmanjaro : https://github.com/manjaro/packages-community/pull/97#issuecomment-169744863

**Comment by philmmanjaro** [comment]
@oberon2007: the given theme is not **evopop** at all. It is a fork from bunsenlabs. They even don't maintain it anymore. **openbox** should simply adopt to a given working gtk 3.18 theme which is currently maintained. Also adding fonts and themes to the profile overlay is a no-go in general:

```
160K	./usr/share/backgrounds
2,4M	./usr/share/themes
196K	./usr/share/slim
8,00K	./usr/share/grub
2,1M	./usr/share/fonts
24,0K	./usr/share/icons
```

The problem was, we never controlled these community profiles in general before we merge them.

**Comment by philmmanjaro** [comment]
@udeved: waiting another week to sort it out won't do anything. It rather push people still to add new stuff to **master**. I rather suggest to merge sooner than later. To have a "clean" master, I'd suggest to hold those profiles back, which we have to analyse further.

**Comment by udeved** [comment]
I don't mind merge today and ban the bad profiles until sorted out.

**Comment by philmmanjaro** [comment]
I'm totally fine with that. This also shows that we have some kind of standard already ;)

**Comment by philmmanjaro** [comment]
I'd suggest to fix the **sonar** profiles first. Also testing them would be great.

**Comment by udeved** [comment]
I would fix  sonar if I knew how to do that with binary dconf blob. I am generally not really skilled with gtk.

**Comment by philmmanjaro** [comment]
For now merge it as it is. I'll take care of the rest. I just want the other maintainers to proper their profiles before we merge them, otherwise this will go on and on and no changes ...

**Comment by udeved** [comment]
Ok, will merge in a few minutes then.

**Comment by philmmanjaro** [comment]
Sonar profiles will be analysed with this task: https://github.com/manjaro/manjaro-tools-iso-profiles/issues/96

**Comment by udeved** [comment]
Ok, I think I removed the conflicts after merge, will double check now and then push to master.

**Comment by philmmanjaro** [comment]
Sure, create a 15.12 branch before you merge master as usual.

**Comment by udeved** [comment]
Already did 15.12 branch in the new year, iirc 2 days ago

**Comment by udeved** [comment]
https://github.com/manjaro/manjaro-tools-iso-profiles/commit/4af27add4dbe8c2ebb199f7733461e395f17859e

Done.

**Comment by philmmanjaro** [comment]
Great. Then release the new **manjaro-tools** and announce the change. All maintainers with bigger profiles should analyse them and reduce their size. Some topics we have already ;)

**Comment by udeved** [comment]
Yes, I was gonna try to fix manjaro-chroot issue tonight before release.

**Comment by philmmanjaro** [comment]
Ok, that makes totally sense ;)

**Comment by udeved** [comment]
Btw https://github.com/manjaro/manjaro-tools/issues/171
 phil, does any installer not create a /boot partition when auto partitoning?

**Comment by philmmanjaro** [comment]
**Thus** does, **Calamares** not.

**Comment by udeved** [comment]
No good I think, manjaro-chroot relies on os-prober, which seems to expect /boot.

**Comment by philmmanjaro** [comment]
Hmm, should be not the problem as having no extra /boot partition is sometimes normal. I've to check that further if needed. However you can take a look at the [install logs](http://sourceforge.net/projects/manjarolinux/files/release/15.12/manjaro-15.12-logs.zip) of 15.12 release if you like. I used autopartitioning.

**Comment by udeved** [comment]
No problem, there is still a little bug to fix.
But, if the problem persists after the bug was fixed, it would be kind of worrying, since os-prober wouldn't find a manjaro that was autopartitioned with grub either.

**Comment by philmmanjaro** [comment]
Yeah. There will be always some bug ;)

**Comment by oberon2007** [comment]
Well, **os-prober** does definitely find the installation in question! How else would **update-grub** succeed?

**Comment by philmmanjaro** [comment]
Exactly. I also don't think it is within **os-prober** ...

**Comment by philmmanjaro** [comment]
@udeved: maybe compare **mhwd-chroot** with your code and see what the difference is ...

**Comment by udeved** [comment]
I am talking about a second linux, dual or triple boot systems @oberon2007 

It would occur only if the os-prober kicks in eg with grub to detect other OSes, which is done after the local running sytsem it does not check.

**Comment by udeved** [comment]
I have setup a vbox triple boot, all 3 manjaro. We will see.
manjaro-chroot works fine if there is only one linux installed.

**Comment by udeved** [comment]
So any volunteers to unify the backgrounds for mate?

We still don't know any download source of jwm stuff @holmeslinux

**Comment by philmmanjaro** [comment]
For mate you have to build the ISO. Load it to a virtual machine you can either share files with your desktop or you upload the dconf binary to a filehoster. So the changes have to been made in a clean install to have them minimal as possible. Then put it together in the profile and recreate the image for testing the result. Mostly it is fine to just remove the desktop-image folder and recreate the ISO with the -c switch. Normally a job of the profile maintainer.

**Comment by udeved** [comment]
Do you see the binary blob insanity just for changing a background path?
What used to be xml with gconf?

**Comment by philmmanjaro** [comment]
If you change a path of the background or even select a different one you have to do it within the dconf binary. Sorry, but this is the new way to go. For more information [read this](http://www.linux.org/threads/dconf-explained.7764/). Just one quote to satisfy you: `Dconf provides a Windows-registry-like interface. ~/.config/dconf/user is the binary database used by Dconf.` Hope this makes it clear.

**Comment by udeved** [comment]
Even MS allows to disassemble .net code, but this is crazy.
The should at least provide an editor or import function to the dconf.
I can just shake my head what genius pushed this through.

**Comment by philmmanjaro** [comment]
Canonical Ltd was the genius. Also there is an editor called **dconf-editor**. A [how-to](http://askubuntu.com/questions/22313/what-is-dconf-what-is-its-function-and-how-do-i-use-it) is also available. You can also read the [mate docu](http://wiki.mate-desktop.org/docs:gsettings). To add a new wallpaper you simply do `gsettings set org.mate.background picture-filename '/usr/share/backgrounds/mate/abstract/Spring.png'` or point to any other picture you like. However you have to add the new dconf binary generated out of it to your profile or execute the gsettings cmd during creation within a script, which is silly.

**Comment by coffeeking** [comment]
hi
Unlike the windows registry, it can be replaced easily without ill 
effect. If it gets hopelessly messed up, which has happened on me a few 
times, simply remove that file and it will be regenerated. You will have 
to reset some of your settings, but this is better than your system 
being unbootable. Afaik the windows registry stores nearly everything 
about a computer, including the hardware installed in it. Delete those 
files and suddenly it can't boot
Thanks
Kendell clark

**Comment by udeved** [comment]
I have read enough to not want to fiddle with it.

**Comment by oberon2007** [comment]
There is a better way to deliver default gsettings for an ISO than **dconf** on user level!
As I already said: This can be done quite elegantly by means of overrides in **usr/share/glib-2.0/schemas**. For deepin I do that with a package, like that: https://github.com/oberon2007/deepin-desktop-schemas-manjaro/tree/master/overrides Ther you can define everything, be it the color scheme of mousepad, icon size of nautilus ... ...

**Comment by oberon2007** [comment]
The binary is then compiled during package installation: https://github.com/manjaro/packages-community/blob/master/deepin-manjaro/deepin-desktop-schemas-manjaro/deepin-desktop-schemas-manjaro.install#L2

**Comment by oberon2007** [comment]
It is tedious in the beginning to go through all the gschemas, but once you know where you find what you can edit everything quite easily without having to be inside a built system.

**Comment by coffeeking** [comment]
hi
Burnhard, would you mind doing something similar with sonar's settings? 
One for gnome and one for mate? This is something completely over my 
head, or I'd attempt this myself. I'd love to not have to mess with 
dconf binaries, and It would be nice to be able to add/change shortcuts, 
desktop icons, change themes, etc and then roll this out to users as an 
update, instead of making them re-install their entire OS to get new 
changes.  This would be in line, I think, with issue #95. At least, I 
think that's the point of that issue. If you would I'd really, really 
appreciate it. Would you like to co-maintain espeak with me? If I have 
build problems I can come to you, but you need not handle new versions, 
dependency changes, etc as I can handle little stuff like that.
Thanks
Kendell clark

**Comment by philmmanjaro** [comment]
@oberon2007: this makes totally sense and should be adopted for #95 and #96. Will take a look at it on how to implement this properly. With a new package users might be able to adopt to new changes, however, this has to be verified first, as we shouldn't temper with user settings at all.

**Comment by udeved** [comment]
This means the aim is to provide the xml schema files?

**Comment by oberon2007** [comment]
@philmmanjaro This will only define the **default** settings for the system. Once the user has made changes to the gsettings the gsettings schemas will not interfere, afaik. It's like what you see printed in bold in dconf-editor and what will be applied when you hit 'default' there.

**Comment by oberon2007** [comment]
Exactly, @udeved . We would provide the "vendor override files" https://developer.gnome.org/gio/stable/glib-compile-schemas.html

**Comment by udeved** [comment]
How would this work out for eg mate, the mate packages provide schemas, right?

**Comment by oberon2007** [comment]
We'll have to check out what is present in the Mate build. An override file can be built from dedicated gsettings files like that: https://github.com/oberon2007/deepin-desktop-schemas-manjaro/blob/master/Makefile#L7

**Comment by udeved** [comment]
I see, so schemas would end up in the desktop overlay.
This is better than nothing and just binary blob, we just need to be careful with filesystem conflicts.
And I don't know what happens if a pretty complex schem file is borked, that is used to validate or general valid xml.
Currently building mate iso....

**Comment by oberon2007** [comment]
Yes, it's a problem. In Deepin, when you have a faulty path somewhere in gsettings whole parts of the desktop breaks completely - hooray!! :P

**Comment by oberon2007** [comment]
:) good night!

**Comment by oberon2007** [comment]
**/usr/share/glib-2.0/schemas/org.gnome.desktop.background.gschema.xml** :P ;)

**Comment by philmmanjaro** [comment]
For mate it is **org.mate.background**, however I recommend to go with the binary for now.

**Comment by philmmanjaro** [comment]
Maybe there is also a **org.mate.desktop.background** somewhere. We have to check that in the editor first.

**Comment by oberon2007** [comment]
Yes, it is **org.mate.background**, owned by **mate-desktop-gtk3**
I agree that for now dconf is an easier fix, especially when it's not your own profile. To do this properly one would have to start from a default mate install and see what else needs to be adjusted with an override. Of course you can just delete user level dconf and then see what the desktop looks like, but we will not know what the maintainer originally might have set apart from the wallpaper ... ...
Anyway I still don't understand why this wallpaper is in three different places ;) The desktop uses **/usr/share/backgrounds/mate/desktop/Stripes.png**

**Comment by oberon2007** [comment]
which is the monster, unfortunately ;)

**Comment by philmmanjaro** [comment]
What happens by simply removing the others. All this has to been verified. I also think a jpg would be fine in a smaller resolution. However, wallpapers should be provided in a package anyway.

**Comment by oberon2007** [comment]
Well hang on! Before we fumble any longer with this profile... This is totally NOT the ISO, ringo has published! Different wallpaper to begin with. There is not even a menu!!

**Comment by philmmanjaro** [comment]
Then we have to talk to Ringo which sources he used and that he should update the profile accordingly.

**Comment by oberon2007** [comment]
absolutely. What's his github alias btw??

**Comment by udeved** [comment]
https://github.com/ringo32

This one I believe.

**Comment by oberon2007** [comment]
absolutely ;) @ringo32 , help please! Could you push your up-to-date profiles?

**Comment by udeved** [comment]
Would 1.5MB for a profile an acceptable limit to be adopted in community?

**Comment by oberon2007** [comment]
@coffeeking , sorry for not answering your questions directly! I'll be happy to assist you, and @philmmanjaro is also there for you :) I just quickly attempted to build **espeak-ng** yesterday and it failed about a Russian plugin I think. We'll need the wizard again, I think ;)
For the profiles I'd think it would be ideal to base the sonar profiles on the existing community ones and to spot the typical and necessary differences and additions for your versions - not sure, yet how to best investigate this... First of all we'll need the latest community profiles to start with ;)

**Comment by Esclapion** [comment]
What's the interest to shorten so much the profiles ? A size of 5-10 MB is absolutely acceptable, imho.

**Comment by udeved** [comment]
5-10 MB is too big, the git repo is way too big, and contains stuff that should be packaged.

lxqt is fine with 512 kB

**Comment by philmmanjaro** [comment]
@Esclapion: we have some unwritten rules:

* don't add wallpapers
* don't add fonts
* don't add themes

These have to be provided via a package. Each binary we ship we also add to our git-overlay, which is currently at 170 MB if you download our git-repo the normal way for contribution.

**Comment by udeved** [comment]
Rob said, the enlightenment theme can be removed in favor of a displaymanager.
I would propose lightdm?

**Comment by Esclapion** [comment]
But it's too long to download or what ? If we have 20 profiles at 5 MB, it means 100 MB. 20" ?

Else, i'm for having clean profiles, but some like enlightenment are difficult to reduce, by nature.

**Comment by philmmanjaro** [comment]
Current mate profile can be found [here](https://github.com/manjaro-nl/manjaro-tools-iso-profiles/tree/16.dev1).

**Comment by udeved** [comment]
See #101

**Comment by philmmanjaro** [comment]
@udeved: please convert [current profiles](https://github.com/manjaro/manjaro-tools-iso-profiles/tree/manjaro-nl-dev) to v0.10. Branch can be deleted afterwards. Thx.

```
768K	./cinnamon
928K	./mate
```
